### PR TITLE
Added RPC command line arguments

### PR DESCRIPTION
**Pull request by andrasfuchs**
Added some RPC-related command line arguments and modified the argument handling.

So now you can connect to Bitcoin and Stratis full nodes with RPC or Stratis with REST API.

Here are some examples:
`dotnet run stratis Wallet/history WalletName=testwallet - Lists all the historical transactions of the wallet called 'testwallet'.`  
  
`dotnet run stratis getinfo -rpcuser=stratistestuser -rpcpassword=stratistestpassword -rpcconnect=127.0.0.3 -rpcport=26174 - Displays general information about the Stratis node on the 127.0.0.3:26174, authenticating with the RPC specified user.`  
  
`dotnet run bitcoin getbalance -rpcuser=btctestuser -rpcpassword=btctestpass - Displays the current balance of the opened wallet on the 127.0.0.1:8332 node, authenticating with the RPC specified user.`

### CODE REVIEW

**Comment by bokobza** [comment]
Looks very interesting. Can you resolve the conflicts first please :-) ?

**Comment by andrasfuchs** [comment]
Sorry, I messed this one up a little, because I forgot to pull your master branch before posting the pull request. Hopefully it's ok now :)

File: src/Stratis.Bitcoin.Cli/Program.cs [hunk 1]
```diff
@@ -8,72 +9,143 @@
...
+                List<string> commandArgList = new List<string>(argList);
+
                 // Display help if required.
-                if (args.Length == 0 || args.Contains("-help") || args.Contains("--help"))
+                if (command == null || argList.Contains("-help") || argList.Contains("--help"))
                 {
                     var builder = new StringBuilder();
                     builder.AppendLine("Usage:");
-                    builder.AppendLine(" dotnet run <Stratis.Bitcoin.Cli/Stratis.Bitcoin.Cli.dll> [network-name] [command] [arguments]");
+                    builder.AppendLine(" dotnet run <Stratis.Bitcoin.Cli/Stratis.Bitcoin.Cli.dll> [network-name] [options] <command> [arguments]");
```
Comment (bokobza) [inline src/Stratis.Bitcoin.Cli/Program.cs]: So do we start the CLI by executing a command directly?
It's not `dotnet run` and then we pass commands?

File: src/Stratis.Bitcoin.Cli/Program.cs [hunk 2]
```diff
@@ -8,72 +9,143 @@
...
-                        NodeSettings nodeSettings = new NodeSettings(blockchain, network).LoadArguments(args);
+                        string[] options = optionList.ToArray();
+
+                        // TODO: NetworksContainer doesn't contain the Stratis networks, so the default port number will be not retrieved here
+                        Network network = Network.GetNetwork(networkName);
+
+                        NodeSettings nodeSettings = new NodeSettings(blockchain, network).LoadArguments(options);
+
                         var rpcSettings = new RpcSettings();
+                        // read the values from the configuration file
```
Comment (bokobza) [inline src/Stratis.Bitcoin.Cli/Program.cs]: please leave an extra line above the comment.

File: src/Stratis.Bitcoin.Cli/Program.cs [hunk 3]
```diff
@@ -8,72 +9,143 @@
...
+                            rpcURI = new Uri($"http://{rpcAddress}:{rpcPort}");
+                        }
+                        rpcSettings.RpcUser = options.GetValueOf("-rpcuser") ?? rpcSettings.RpcUser;
+                        rpcSettings.RpcPassword = options.GetValueOf("-rpcpassword") ?? rpcSettings.RpcPassword;
+
+                        Console.WriteLine($"Connecting to the following RPC node: http://{rpcSettings.RpcUser}:{rpcSettings.RpcPassword}@{rpcURI.Authority}...");
 
-                        // Initialize the RPC client with the configured or passed userid, password and endpoint
-                        RPCClient rpc = new RPCClient($"{rpcSettings.RpcUser}:{rpcSettings.RpcPassword}", new Uri($"http://{nodeEndPoint}"));
+                        // Initilize the RPC client with the configured or passed userid, password and endpoint.
```
[the remark this answers is not in the document]
Comment (andrasfuchs) [reply src/Stratis.Bitcoin.Cli/Program.cs]: @bokobza as you recommended before I recreated this pull request here: https://github.com/stratisproject/StratisBitcoinFullNode/pull/793, so please ignore this one.

**Pull request closed without merging**